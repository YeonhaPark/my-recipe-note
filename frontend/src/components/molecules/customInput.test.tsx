import { mount } from 'enzyme';
import { Input } from '.';
import renderer from 'react-test-renderer';

const InputProps = {
  type: 'password',
  labelName: 'Password',
  placeholder: 'Password',
  required: true,
};

const renderInput = () => mount(<Input {...InputProps} />);

describe('Input', () => {
  it('renders Input', () => {
    renderInput();
  });

  it('renders exact prop provided', () => {
    const input = renderInput();

    expect(input.prop('required')).toEqual(true);
  });

  it('changes required prop value', () => {
    const customInput = renderInput();
    customInput.setProps({ required: false });
    expect(customInput.prop('required')).toEqual(false);
  });

  it('matches snapshot', () => {
    const tree = renderer.create(<Input {...InputProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
