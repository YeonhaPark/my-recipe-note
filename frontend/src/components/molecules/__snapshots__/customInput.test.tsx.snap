// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input matches snapshot 1`] = `
<div
  className="css-hqqaza-Input"
>
  <label
    className="MuiInputLabel-root MuiInputLabel-animated MuiFormLabel-root MuiFormLabel-colorPrimary css-9npbnl-MuiFormLabel-root-MuiInputLabel-root"
    htmlFor="Password"
  >
    Password
  </label>
  <div
    className="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth css-wjdba9-MuiInputBase-root-MuiInput-root"
    onClick={[Function]}
    required={true}
  >
    <input
      className="MuiInput-input MuiInputBase-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
      id="Password"
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="Password"
      required={true}
      type="password"
    />
  </div>
</div>
`;
